src/components/comment-list/CommentList.jsx

```diff
@@ -4,16 +4,70 @@ import { getComments } from "../../utils/api";
 import { formatDate } from "../../utils/helper";
 import { CommentForm } from "../comment-form/CommentForm";
 import { Loader } from "../loader/Loader";
+import { deleteComment } from "../../utils/api";
 
-export function CommentList({ articleId }) {
+export function CommentList({ articleId, user }) {
   const [comments, setComments] = useState([]);
+  const [deleteCommentId, setDeleteCommentId] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [successMessage, setSuccessMessage] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
 
   useEffect(() => {
-    getComments(articleId).then(({ comments }) => {
-      setComments(comments);
-    });
+    setLoading(true);
+    getComments(articleId)
+      .then(({ comments }) => {
+        setComments(comments);
+        setLoading(false);
+      })
+      .catch((err) => {
+        console.log("error:", err);
+        setErrorMessage("there was a problem fetching the comments");
+        setLoading(false);
+      });
   }, []);
 
+  const handleDeleteCross = (commentId) => {
+    setDeleteCommentId(commentId);
+  };
+
+  const handleConfirmDelete = (commentId) => {
+    setErrorMessage("");
+    setSuccessMessage("");
+    setLoading(true);
+
+    //optimistic UI update
+    const updatedComments = comments.filter(
+      (comment) => comment.comment_id !== commentId
+    );
+    setComments(updatedComments);
+
+    deleteComment(commentId)
+      .then(() => {
+        // should I make another api call to refetch comments
+        // or should I use setComments somehow without fetching
+        // should I now include comments in the dependency arr?
+        //I want to display new list of comments after deleting
+        // and at the same time get rid of the modal
```

Comment:
So you can achieve this logic without optimistic rendering. You can refetch the comments once the delete button has been pressed by giving the useEffect a dependency to track. This could even be something like a boolean called `isDeleting`